### PR TITLE
[WFLY-16542]:Remove the xerces dependency from WS deployment for Wild…

### PR DESCRIPTION
**Pull request by jimma**
…Fly Preview

https://issues.redhat.com/browse/WFLY-16542

### CODE REVIEW

File: webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSDependenciesProcessor.java [hunk 1]
```diff
@@ -66,8 +67,11 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
...
             moduleSpec.addSystemDependency(new ModuleDependency(moduleLoader, api, false, false, true, false));
         }
-        //After jboss modules 2.0, the xercers module is not added
-        moduleSpec.addSystemDependency(new ModuleDependency(moduleLoader, XERCES_IMPL, true, false, true, false));
 
+        //After jboss modules 2.0, the xerces module is not added and this is required for jaxb 2.3.x(EE8) to pass the TCK
+        //But this is not needed for jaxb 3.x in WildFly Preview(EE10)
+        if (Endpoint.class.getPackage().getImplementationVersion().startsWith("3")) {
+            moduleSpec.addSystemDependency(new ModuleDependency(moduleLoader, XERCES_IMPL, true, false, true, false));
+        }
```
Comment (jamezp) [inline webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSDependenciesProcessor.java:75]: I almost wonder if we should just remove this. Once we merge the big-bang change, 2.3.x won't be used anyway.

Comment (jimma) [reply webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSDependenciesProcessor.java:75]: @jamezp Will there be any branch/tag before the big-bang change ?

Comment (bstansberry) [reply webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSDependenciesProcessor.java:75]: @jimma We will create a tag before merging the big-bang branch.

Comment (jamezp) [reply webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSDependenciesProcessor.java:75]: Ah, I say we keep it then so we can get a good tag before we remove it after the big-bang.

**Pull request merged**